pkg/eventcollector/eventcollector.go

```diff
@@ -121,11 +126,26 @@ func (c *EventCollector) handleEvent(payload interface{}) interface{} {
 
 // StartCollection contains logic to run retriever and listener.
 func (c *EventCollector) StartCollection() error {
-	err := c.retrieveEvents()
+	err := c.retrieveEvents(c.filterers)
 	if err != nil {
 		return fmt.Errorf("Error retrieving events: err: %v", err)
 	}
 	pastEvents := c.retrieve.PastEvents
+
+	// Check pastEvents for any new newsrooms to track
+	err = c.CheckRetrievedEventsForNewsroom(pastEvents)
+	if err != nil {
+		return fmt.Errorf("Error checking newsroom events during filterer, err: %v", err)
+	}
+	if len(c.filterTriggers) > 0 {
+		// NOTE(IS): This overwrites the previous retriever with the new filterers
```

Comment:
This part kind of sucks because it overwrites the previous retriever, but we don't really need it anymore so maybe it's ok?

Reply:
This can probably wait until a better time, but I was thinking here was to update `retrieveEvents` so it just returns the `pastEvents` without setting it as a property, or perhaps returns the entire `EventRetriever`.  In the latter case, we can then set it or just use the results in this function and we won't have this weird hacky issue.